fhir-search/src/test/java/com/ibm/fhir/search/test/InclusionParameterParseTest.java

```diff
@@ -327,4 +340,151 @@ public void testMultiIncludeRevinclude() throws Exception {
+    }
+
+    @Test
+    public void testWildcardRevIncludeSingleMatchingTargetType() throws Exception {
+        Map<String, List<String>> queryParameters = new HashMap<>();
+        Class<Encounter> resourceType = Encounter.class;
+
+        queryParameters.put("_revinclude", Collections.singletonList("MedicationAdministration:*:Encounter"));
+        FHIRSearchContext searchContext = SearchUtil.parseQueryParameters(resourceType, queryParameters);
+
+        assertNotNull(searchContext);
+        assertTrue(searchContext.hasRevIncludeParameters());
+        assertEquals(1, searchContext.getRevIncludeParameters().size());
+        InclusionParameter incParm = searchContext.getRevIncludeParameters().get(0);
+        assertEquals("MedicationAdministration", incParm.getJoinResourceType());
+        assertEquals("context", incParm.getSearchParameter());
+        assertEquals("Encounter", incParm.getSearchParameterTargetType());
+        assertFalse(searchContext.hasIncludeParameters());
+
+        String selfUri = SearchUtil.buildSearchSelfUri("http://example.com/Encounter", searchContext);
+        assertTrue(selfUri.equals("http://example.com/Encounter?_count=10&_revinclude=" + incParm.getJoinResourceType() +
+            ":" + incParm.getSearchParameter() + ":" + incParm.getSearchParameterTargetType() + "&_page=1"));
+    }
+
+    @Test
+    public void testWildcardRevIncludeNoTargetType() throws Exception {
+        Map<String, List<String>> queryParameters = new HashMap<>();
+        Class<Patient> resourceType = Patient.class;
+        String include1 = "&_revinclude=Procedure:patient:Patient";
+        String include2 = "&_revinclude=Procedure:performer:Patient";
+        String include3 = "&_revinclude=Procedure:subject:Patient";
+
+        List<InclusionParameter> expectedIncludeParms = new ArrayList<>();
+        expectedIncludeParms.add(new InclusionParameter("Procedure", "patient", "Patient"));
+        expectedIncludeParms.add(new InclusionParameter("Procedure", "performer", "Patient"));
+        expectedIncludeParms.add(new InclusionParameter("Procedure", "subject", "Patient"));
+
+        queryParameters.put("_revinclude", Collections.singletonList("Procedure:*"));
+        FHIRSearchContext searchContext = SearchUtil.parseQueryParameters(resourceType, queryParameters);
+
+        assertNotNull(searchContext);
+        assertTrue(searchContext.hasRevIncludeParameters());
+        assertEquals(expectedIncludeParms.size(), searchContext.getRevIncludeParameters().size());
+        for (InclusionParameter includeParm : expectedIncludeParms) {
+            assertTrue(expectedIncludeParms.contains(includeParm));
+        }
+
+        assertFalse(searchContext.hasIncludeParameters());
+
+        String selfUri = SearchUtil.buildSearchSelfUri("http://example.com/Patient", searchContext);
+        assertTrue(selfUri.contains(include1));
+        assertTrue(selfUri.contains(include2));
+        assertTrue(selfUri.contains(include3));
+    }
+
```

Comment:
Having an integration test in SearchTest.java in fhir-server-test would be ideal

Reply:
Test added.